plugins/UsersManager/Controller.php

```diff
@@ -387,37 +387,67 @@ public function addNewToken()
             throw new Exception('Not allowed');
         }
 
-        $noDescription = false;
+        $invalidExpireDate = null;
+        if (!empty($_POST['token_expire_date'])) {
+            $tokenExpireDate = \Piwik\Request::fromRequest()->getStringParameter('token_expire_date');
+
+            try {
+                $invalidExpireDate = true;
+                if (preg_match('/^(\d{4})-(\d{2})-(\d{2})$/', $tokenExpireDate)) {
+                    Date::factory($tokenExpireDate);
+                    $invalidExpireDate = false;
+                }
+            } catch (Exception $e) {
+                // nop
+            }
+        }
 
+        $noDescription = null;
+        $description = '';
         if (!empty($_POST['description'])) {
-            Nonce::checkNonce(self::NONCE_ADD_AUTH_TOKEN);
-
             $description = \Piwik\Request::fromRequest()->getStringParameter('description', '');
+            $noDescription = false;
+        } elseif (isset($_POST['description'])) {
+            $noDescription = true;
+        }
```

Comment:
Can't that be simplified in some way? If I see that correctly the variable is used to be passed through the template to vue.js. The vue component assumes the value to be passed as boolean, so passing the default `null` value, that is set, if the POST parameter is missing, might actually result in unexpected behavior.

Reply:
fair point, updated and simplified